plugins/events/create.js

```diff
@@ -22,32 +22,32 @@ module.exports = () => ({
             }
         },
         handler: (request, reply) => {
-            const eventFactory = request.server.app.eventFactory;
-            const pipelineFactory = request.server.app.pipelineFactory;
-            const userFactory = request.server.app.userFactory;
-            const buildFactory = request.server.app.buildFactory;
-            const jobFactory = request.server.app.jobFactory;
+            const { buildFactory, jobFactory, eventFactory, pipelineFactory,
+                userFactory } = request.server.app;
+            const { buildId, causeMessage, creator, meta, parentBuilds } = request.payload;
+            const { scmContext, username } = request.auth.credentials;
             const scm = eventFactory.scm;
-            const scmContext = request.auth.credentials.scmContext;
-            const username = request.auth.credentials.username;
             const isValidToken = request.server.plugins.pipelines.isValidToken;
-            const meta = request.payload.meta;
-            const causeMessage = request.payload.causeMessage;
-            const creator = request.payload.creator;
             const updateAdmins = request.server.plugins.events.updateAdmins;
 
             return Promise.resolve().then(() => {
-                const buildId = request.payload.buildId;
-
                 if (buildId) { // restart case
                     return buildFactory.get(buildId)
                         .then(b => jobFactory.get(b.jobId)
-                            .then(j => ({
-                                pipelineId: j.pipelineId,
-                                startFrom: j.name,
-                                parentBuildId: b.parentBuildId,
-                                parentEventId: b.eventId
-                            })));
+                            .then((j) => {
+                                const restartConfig = {
+                                    pipelineId: j.pipelineId,
+                                    startFrom: j.name,
+                                    parentBuildId: b.parentBuildId,
+                                    parentEventId: b.eventId
+                                };
+
+                                if (parentBuilds) {
+                                    restartConfig.parentBuildsInfo = parentBuilds;
```

Comment:
can't this just be `parentBuilds` instead of `parentBuildsInfo` ?

Reply:
it's defined above so I needed to use another name

Reply:
can't it be `restartConfig.parentBuilds = parentBuilds` ?

then https://github.com/screwdriver-cd/screwdriver/pull/1960/files#diff-6590031cbab3d825a0a4cd19e5da419bR80 can change to

```
if (parentBuilds)```